node/narwhal/tests/bft_e2e.rs

```diff
@@ -123,3 +124,50 @@ async fn test_quorum_break() {
     // Check the nodes have stopped advancing through the rounds.
     assert!(network.is_halted().await);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_leader_election_consistency() {
+    // The round at which to start checking for leader consistency.
+    const STARTING_ROUND: u64 = 2;
+
+    // The minimum and maximum rounds to check for leader consistency.
+    const MIN_ROUND: u64 = 4;
+    const MAX_ROUND: u64 = 30;
+
+    // Start N nodes, connect them and start the cannons for each.
+    const N: u16 = 4;
+    const CANNON_INTERVAL_MS: u64 = 10;
+    let mut network = TestNetwork::new(TestNetworkConfig {
+        num_nodes: N,
+        bft: true,
+        connect_all: true,
+        fire_cannons: Some(CANNON_INTERVAL_MS),
+        // Set this to Some(0..=4) to see the logs.
+        log_level: Some(1),
+        log_connections: true,
+    });
+    network.start().await;
+
+    // Wait for starting round to be reached
+    let cloned_network = network.clone();
+    deadline!(Duration::from_secs(60), move || { cloned_network.is_round_reached(STARTING_ROUND) });
```

Comment:
shouldn't this test be `#[ignore]`d due to its runtime?

Reply:
The runtime is around 69 seconds on my device, Niklas wanted it to run within a timeframe that's somewhat acceptable as part of the normal tests

Reply:
Can we get it below 60 without loosing too much quality?

Reply:
We could probably set the `MAX_ROUND` to 20 instead of 30, that should cut down test times quite a bit (well, by 1/3rd). But that would also mean we're effectively only evaluating 8 rounds.